platform/ios/src/MGLAnnotationView.h

```diff
@@ -16,6 +18,14 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier;
 
 /**
+ This property will be set to the associated annotation when the view is visible.
+ 
+ When the view is queued and waiting to be reused, the value will be set to nil.
+ This property should not be changed directly.
+ */
+@property (nonatomic, strong, nullable) id <MGLAnnotation> annotation;
```

Comment:
I wonder if the reason MapKit leaves this property read-write is that a subclass might want to override `-setAnnotation:` to do something special. At least, that’s the reason MapKit exposes `-setSelected:animated:` publicly with a note saying never to call it (as seen in #5297).